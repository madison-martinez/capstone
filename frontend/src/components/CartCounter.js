import React from "react";
import { useSelector } from "react-redux";
import styled from "styled-components";

const Counter = styled.div`
  background: ${(props) => props.theme.darkBlue};
  color: ${(props) => props.theme.offWhite};
  border-radius: 50%;
  padding: 0.5rem;
  height: 50px;
  display: block;
`;

const CartCounter = () => {
  const cartCount = useSelector(state => state.cart.cartCount)
  
  return (
    <>
    {cartCount > 0 && <Counter>{cartCount}</Counter>}
    </>
  ) 
};



export default CartCounter;
