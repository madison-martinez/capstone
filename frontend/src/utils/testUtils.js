import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { reducers } from '../reducers';

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState ),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

//mock middleware functions
const thunk = ({ dispatch, getState }) => next => action => {
    if (typeof action === 'function') {
      return action(dispatch, getState)
    }
    return next(action)
};

const fakeItem = () => ({
    id: '1',
    price: 20.00,
    title: 'Roses',
    description: 'Roses are Red',
});

export {
    fakeItem,
    thunk
};

export * from '@testing-library/react';
export { render };