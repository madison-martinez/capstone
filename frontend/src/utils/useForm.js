import { useState } from 'react';



export default function useForm(initialState = {}) {
    const [values, setValues] = useState(initialState);

    function handleChange(e) {
        let { value, name, type } = e.target;
        if (type === 'number') {
            value = parseInt(value);
        }
        if (type === 'file') {
            [value] = e.target.files;
        }
        setValues({
            ...values,
            [name]: value
        });
    }

    function clearForm() {
        const emptyForm = Object.fromEntries(
            Object.entries(values).map(([key]) => [key, ''])
        );
        setValues(emptyForm);
    }

    function setInitialForm() {
        setValues(initialState);
    }

    return {
        values, 
        handleChange, 
        clearForm, 
        setInitialForm
    }
}