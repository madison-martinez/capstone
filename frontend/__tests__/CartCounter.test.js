import { render, screen } from "../src/utils/testUtils";
import CartCounter from "../src/components/CartCounter";

describe("<CartCounter />", () => {
  it("should display an empty div when nothing in cart", () => {
    const { container } = render(<CartCounter />);
    expect(container).toMatchSnapshot();
  });

  it("handles state updating to add one to cart", async () => {
    render(<CartCounter />, { initialState: { cart: { cartCount: 1 } } });
    expect(screen.getByText(1)).toBeInTheDocument();
  });
});
