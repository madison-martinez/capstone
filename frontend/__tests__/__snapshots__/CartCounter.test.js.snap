// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCounter /> should display an empty div when nothing in cart 1`] = `<div />`;
